### DISCUSSION TITLE
Tailwind Animation utilities are not working

### DISCUSSION THREAD

**Discussion by thm1d** [tailwindlabs/tailwindcss#2945] [Help] [answered]
I am trying to create a spin component but it's not animating.
here is my code 
```
<div class="relative flex justify-center items-center h-3">
    <div class="rounded-full animate-spin ease duration-300 w-5 h-5 border-2 border-white"></div>
</div>
```

**Comment by chr-ge** [comment]
I copied your code into tailwind play and its spinning. https://play.tailwindcss.com/iKd6C9dhPj
You're trying to animate the spin correct?

**Reply by thm1d** [reply]
Yes, I was trying to make a spinner. There was a problem with my tailwind installation. I then used the CDN build and it started spinning.

**Comment by simonswiss** [comment]
Hey!

Your element might be actually spinning, but you can't see it because of the `rounded-full` class 😅

Here's the same code with `rounded` instead of `rounded-full` (+ some background color and different height), as you can see the element with `animate-spin` is spinning 👍

https://play.tailwindcss.com/Zo6XeEp1bP

**Reply by thm1d** [reply]
There was a problem with my tailwind installation. I then used the CDN build and it started spinning.

**Comment by thm1d** [comment, marked as answer]
I used the CDN build and changed the code to this and it worked :
```
<div wire:loading class="absolute top-0 right-0 mr-2 mt-2">
  <svg class="animate-spin h-4 w-4 rounded-full bg-transparent border-2 border-transparent border-opacity-50" style="border-right-color: white; border-top-color: white;" viewBox="0 0 24 24"></svg>
</div>
```

**Reply by tom-oroshi** [reply]
I was having the same issue with pulse, using CDN worked for me as well. Thanks for posting this.

**Reply by PhilippSchreiber** [reply]
Using the CDN version for fixing this is not a fix, tbh. It's just throwing the whole idea of tailwind away, because you loose the benefits of having only the classes of your code inside of the CSS:

**Reply by PhilippSchreiber** [reply]
For me applying "absolute" to the element with "animate-spin" solved the problem.

**Comment by JacquesvanWyk** [comment]
This might not be answer for all but i ran the command npm run production and that stripped all the purged all the unused classes but when i ran npm run dev all came back.

**Comment by tesla-oss** [comment]
I had the same problem just a moment ago.
In my case, the existing incorrect class name("text‑white" instead of "text-white").
So I solved the problem by putting the tailwind classes in the SCSS styles file, and plugging that class in the component, and only then did I get the warning.
![image](https://github.com/tailwindlabs/tailwindcss/assets/55257833/6c70ee1e-5b1b-4cb8-9c6a-93bdb929a216)
![image](https://github.com/tailwindlabs/tailwindcss/assets/55257833/da516b8c-7a99-4282-8273-acfaa81e3469)

**Comment by sgwanlee** [comment]
in my case, I put custom animation not in the `extend` of `tailwind.config.ts` file
After I put custom animation in the `extend`, the problem solved.

before
```
const config = {
    ...
    extend: {...}
    animation: {
        "loop-scroll": "120s linear 0s infinite normal none running loop-scroll",}
    }
```

fixed
```
const config = {
    ...
    extend: {
         ...,
         animation: {
             "loop-scroll": "120s linear 0s infinite normal none running loop-scroll",}
         }
     }
```

**Reply by dekunle02** [reply]
this worked for me! Thanks

**Reply by voidmozx** [reply]
You are right! Many Thanks.

**Reply by MetaMmodern** [reply]
this was my root cause too, thank you. custom keyframes should also go into "extend"

**Comment by MateusPitura** [comment]
In my case, the problem was that we defined Tailwind directives inside a CSS selector in `index.css`. Removing the surrounding selector made it work, but unfortunately I cannot do that (it's a kind of microfrontend architecture and I need isolation from the main environment). I could add some animations by adding `animation` and `keyframe` to `tailwind.config.js`, but only some of them work.
```css
#root {
  @tailwind base;
  @tailwind components;
  @tailwind utilities;
}
```
Codesandbox: https://codesandbox.io/p/devbox/2x4tz6
